app/main.py

```diff
@@ -1 +1,70 @@
-# write your code here
+class Animal:
+    alive = []  # Class attribute to track all alive animals
+
+    def __init__(self, name):
+        self.name = name
+        self.health = 100
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __del__(self):
+        Animal.alive.remove(self)  # Remove the animal from the alive list when it dies
```

Comment:
Using the `__del__` method to remove an animal from the `alive` list is not reliable, as it depends on the garbage collector to call it. Consider using a different approach to manage the removal of dead animals.